src/commands/install.ts

```diff
@@ -30,43 +42,92 @@ export default class Install extends Command {
     }),
   };
 
-  private async installSingle(service_id: string) {
-    const service_dir = process.cwd();
-    const parts = service_id.split(':');
-    const service_name = parts[0];
-    let service_tag = parts[1];
-    const config = ServiceConfig.loadFromPath(service_dir);
+  async run() {
+    await this.tasks();
+  }
 
-    if (config.name === service_name) {
-      throw new Error('Services cannot depend on themselves');
-    }
+  async tasks() {
+    const { args, flags } = this.parse(Install);
+    const root_service_path = flags.prefix ? flags.prefix : process.cwd();
+    const dependency_graph = await generateGraphFromPaths([process.cwd()], new EnvironmentConfigV1(), this.app.api);
+    const root_service = Array.from(dependency_graph.nodes.values())[0];
 
-    const { data: service } = await this.app.api.get(`/services/${service_name}`);
-    if (!service) {
-      throw new Error(`Service not found in the registry`);
-    }
+    if (args.service_name) {
+      // eslint-disable-next-line prefer-const
+      let [service_name, service_version] = args.service_name.split(':');
+      if (!service_version) {
+        throw new Error(`Please specify a version for ${service_name}`);
+      }
+      if (root_service.name === service_name) {
+        throw new Error('Cannot install a service inside its own config');
+      }
+
+      const new_dependencies: { [s: string]: string } = {};
+      if (new_dependencies[service_name]) {
+        throw new Error(`A version of ${service_name} is already installed.`);
+      }
```

Comment:
This conditional will never be hit since you initialize the object in the line prior